STIPS/STIP-002.md

```diff
@@ -0,0 +1,150 @@
+- Finally we require, newBalance == existingBalance + quantity
+    - LHS = newBalance
+        - = newScaledBalance * index
+        - = (initialScaledBalance + (quantity/index)) * index
+        - = (initialScaledBalance * index) + (quantity/index)*index
+    - RHS = existingBalance + quantity
+        - = (initialScaledBalance * index) + quantity
+- Cancelling, initialScaledBalance * index from LHS and RHS,
+- We essentially require, (quantity/index) * index == quantity, with index > 1
+- Which will not always be true because solidity uint256 has limited precision and there is a chance of `(quantity/index) * index` to be greater/lesser than `quantity` by 1 wei.
+
+## Feasibility Analysis
+
+### Option 1:
+Instead of over-optimizing our checks to test for both undercollaterlization and overcollaterlization. We can just have checks to prevent undercollaterlization. This would prevent reverting in cases when `newBalance` of aToken is greater than `existingBalance + quantity` by 1 wei, and thus decrease the failure rate of issuance/redemption (maybe to 33%, cause we can consider less than, equal to, and greater than to be 3 equal cases). This option is the best considering time. The change is contained within the smart contracts and doesn't bubble up the stack leading to bigger changes in our backend.
+
+
+
+## Open Questions
+
+1. Do we restrict usage of these modified issuance modules (which are linked to the new libraries) from third-party use?
+    - Why did we have the strict checks in the first place?
+        - To make sure the set is not undercollateralized 
+            - Eg. To stop tokens which collects fees on transfer from being a component in a SetToken
+        - This strict check achieved the above requirement, but at the same time it also prevented overcollateralization
+    - What are the only requirements for our system to function well?
+        - We do not want Set to be undercollateralized
+        - We are fine with overcollaterlization (discussed below)
+    - Thus, the new checks full fill our requirements.
+        - So, we need not restrict usage of these modified issuance modules
+
+2. Do we restrict these new modified issuance modules (which are linked to the new libraries) only to SetToken which hold aTokens? Or do we make these issuance modules the default going forward?
+    
+2. Are we fine with Set being overcollaterlized?
+    - Yes, SetToken can resync the extra assets present to increase their default positions with help of modules
+        - ALM and CLM can resync positions for leveraged tokens
+        - Airdrop module can add extra assets as positions for other SetTokens
+    - Also, issuance/redemption generally would not lead to overcollaterlization
+        - For it to happen, we would need the special case of a token which rewards holders on a transfer
+        - we can handle that case using Airdrop module
+
+3. How to handle `newBalance < existingBalance.add(_quantity))`?
```

Comment:
I think this is fine as long as we maintain collateralization. However, note that this means a token which charges a fee upon transfer could in theory be used to get any excess amount of that token held by the Set. Excess amount being defined as a remaining amount of wei from a manager action OR any potential tokens accrued to the Set via farming but not yet "absorbed" into a position via the AirdropModule. This could NOT affect other tokens in the Set though only the token that charges the transfer fees (ie any other token that has not been absorbed into a position could not be stolen.

Reply:
Added to STIP.